### PR TITLE
3/4 Update : Chair and Table Loose ends and fixes

### LINKED ISSUE #17487: [Bug]: Tables built by players become invisible when finished
### Checks

- [X] I have searched the bug with a few keywords, and I confirm this bug was not yet reported.

### Location

Horizon

### Describe the issue

When a table is manually constructed it becomes invisible once it is finished.

### How to reproduce

1. Make table frame.
2. Apply material sheet to finish table.
3. Watch the table become invisible once it is finished.

### Round ID

cp4-bafs

### Staff ckey

_No response_

### PR DESCRIPTION
**Pull request by alsoandanswer**
I just want to sprite, boss...

changes:
  - bugfix: "Fixes constructed tables being invisible."
  - imageadd: "Updates table reinforcement overlays."
  - tweak: "Table frames are now made using two rods."
  - tweak: "Fancy tables are now constructable - use carpet on a table frame."
  - tweak: "Blue carpet renamed and repathed to Cyan carpet for accuracy."
  - bugfix: "Fixes Cyan carpet not having corner sprites."
  - bugfix: "Fixes mapload chair items not setting materials."
  - bugfix: "Fixes fancy chair sprites."
  - bugfix: "Fixes comfy chair name and description."
  - imageadd: "Updates Skrell and Diona tables."
  - backend: "Chairs now use contained sprites."

Fixes #17487

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!bugfix

**Comment by alsoandanswer** [comment]
!review

File: code/modules/materials/materials.dm [hunk 1]
```diff
@@ -57,11 +57,11 @@
 	var/wall_colour                                      // Colour applied to walls specifically.
 	var/icon_base = "solid"                              // Wall and table base icon tag. See header.
 	var/door_icon_base = "metal"                         // Door base icon tag. See header.
-	var/icon_reinf = "reinf_metal"                       // Overlay used
+	var/reinf_icon = "reinf_over"                       // Handles type of overlay used for tables and walls. Don't change without checking every use case.
```
Comment (FluffyGhoster) [inline code/modules/materials/materials.dm:60]: dmdoc the comment

File: code/modules/tables/flipping.dm [hunk 2]
```diff
@@ -1,6 +1,8 @@
 
 /obj/structure/table/proc/straight_table_check(var/direction)
-	if(health > 100)
+	if(material?.weight > DEFAULT_TABLE_FLIP_WEIGHT)
+		return 0
```
Comment (FluffyGhoster) [inline code/modules/tables/flipping.dm:4]:
```suggestion
		return FALSE
```

File: code/modules/tables/flipping.dm [hunk 3]
```diff
@@ -1,6 +1,8 @@
...
+	if(reinforced?.weight > DEFAULT_TABLE_FLIP_WEIGHT)
 		return 0
```
Comment (FluffyGhoster) [inline code/modules/tables/flipping.dm:6]:
```suggestion
		return FALSE
```

File: code/modules/tables/tables.dm [hunk 4]
```diff
@@ -361,19 +369,37 @@
...
+				if(material.icon_colour)
+					I.color = material.icon_colour
 			add_overlay(I)
-			if(material_alteration & MATERIAL_ALTERATION_NAME)
-				name = "[material.display_name] table"
+			if(material.display_name)
+				if(material.display_name == "comfy")
+					name = "fancy table"
+				else
+					name = "[material.display_name] table"
```
Comment (FluffyGhoster) [inline code/modules/tables/tables.dm:390]: Is there any particular reason why this is done in the icon update proc, instead of Initialize()?

Comment (alsoandanswer) [reply code/modules/tables/tables.dm:390]: cause you can remove the material and re-add it again - it's not like a chair where when you deconstruct it and construct it again it despawns and respawns it

**Review by DreamySkrell — APPROVED** [review]

**Pull request merged**